src/main/python/rlbot/botmanager/agent_metadata.py

```diff
@@ -0,0 +1,11 @@
+from rlbot.botmanager.helper_process_request import HelperProcessRequest
+
+
+class AgentMetadata:
+    def __init__(self, index: int, name: str, team: int, pids: set,
```

Comment:
With the number of positional arguments it may be good to consider using `**kwargs` instead.

Reply:
I prefer it like this so the IDE can infer types.

Reply:
There is also the option of using required keyword arguments (which I believe would work with the types as well): https://stackoverflow.com/a/36446353/1581558

```python
def func2(a, *, b):
    ...
```

This is something I didn't know of before, and was apparently defined for Python 3.x in [PEP 3102](https://www.python.org/dev/peps/pep-3102/)

Reply:
I don't think that would improve readability. The current function signature already supports calls like `AgentMetadata(index=self.index, name=self.name, ...` but I didn't bother because the variable names are already obvious.

I don't think we need any of the flexibility that kwargs offers.